app/lnd/lib/neutrino.js

```diff
@@ -79,11 +80,23 @@ class Neutrino extends EventEmitter {
       if (!this.state.chainSyncStarted) {
         const match = line.match(/Syncing to block height (\d+)/)
         if (match) {
-          const height = match[1]
           this.state.chainSyncStarted = true
-          this.state.targetBlockHeight = height
           this.emit('chain-sync-started')
-          this.emit('got-final-block-height', height)
+          // Thhis is the latest block that BTCd is aware of.
+          const btcdHeight = match[1]
+          // Which may not be the actualy current block heighht, so try to fetch that from some block explorers.
+          fetchBlockHeight()
+            // If we were unable to fetchh from bock explorers, then use what BTCd gave us.
+            .catch(err => {
+              mainLog.warn(`Unable to fetch block height: ${err.message}`)
+              return Promise.resolve(btcdHeight)
+            })
+            .then(height => {
+              this.state.targetBlockHeight = height
+              this.emit('got-final-block-height', height)
```

Comment:
They way I have it we will emit the height fetched from the explorer or if none was fetched then the height reported buy btcd will be emitted as a fallback.

This `then` block should have `height` set as `btcdHeight` from `Promise.resolve(btcdHeight)` in the `catch` block above.

Reply:
Yeah sorry, I'm referring to the 2nd `catch`, thoughts on removing that?

Reply:
lint requires it

Reply:
Ah yeah, so removing it would make failure silent, but we should want noisy error so how about just throwing the error there? Or log + throw?

Reply:
We can do, although if you look at the code in the block above, I don't think it is possible that there could be an error since the only thing we are doing is setting a variable and emitting an event. The thing that could have had an  error was dealt with in the previous catch block. I'll add a log entry just for completeness sake...

Reply:
added error message